src/charm.py

```diff
@@ -163,63 +134,85 @@ def _on_install(self, _: InstallEvent) -> None:
 
-        self.unit.status = MaintenanceStatus("Generating configuration files")
+        self.unit.status = MaintenanceStatus("Copying example configuration files")
         copy_files(origin=CONFIG_ORIGIN_PATHS, destination=CONFIG_PATHS)
 
-        self.unit.status = MaintenanceStatus("Generating systemd files")
+        self.unit.status = MaintenanceStatus("Configuring srs env service")
         self._configure_srsenb_service()
-        self._configure_srsue_service()
 
         service_enable(SRS_ENB_SERVICE)
-        self._stored.installed = True
+        self._set_peer_data("installed", True)
 
     def _on_start(self, _: StartEvent) -> None:
         """Triggered on start event."""
+        if not self.unit.is_leader():
+            return
         self.unit.status = MaintenanceStatus("Starting srsenb")
         service_start(SRS_ENB_SERVICE)
-        self._stored.started = True
-        self.unit.status = self._get_current_status()
+        self._set_peer_data("started", True)
+        self.unit.status = ActiveStatus(self._active_status_msg)
 
     def _on_stop(self, _: StopEvent) -> None:
         """Triggered on stop event."""
+        if not self.unit.is_leader():
+            return
         self._reset_environment()
         service_stop(SRS_ENB_SERVICE)
-        self._stored.started = False
-        self.unit.status = self._get_current_status()
+        self._set_peer_data("started", False)
+        self.unit.status = BlockedStatus("Unit is down, service has stopped")
 
     def _on_config_changed(self, _: ConfigChangedEvent) -> None:
         """Triggered on config changed event."""
-        self._stored.bind_addr = self._get_bind_address()
         self._configure_srsenb_service()
-        if self._stored.started:
+        if self._get_peer_data("started"):
             self.unit.status = MaintenanceStatus("Reloading srsenb")
             service_restart(SRS_ENB_SERVICE)
-        self.unit.status = self._get_current_status()
+        self.unit.status = ActiveStatus(self._active_status_msg)
 
     def _on_update_status(self, _: UpdateStatusEvent) -> None:
         """Triggered on update status event."""
-        self.unit.status = self._get_current_status()
+        self.unit.status = ActiveStatus(self._active_status_msg)
+
+    def _on_lte_core_available(self, event: LTECoreAvailableEvent) -> None:
+        """Triggered on lte_core_available.
+
+        Retrieves MME address from relation, configures the srs enb service and restarts it.
+        """
+        if not self.unit.is_leader():
+            return
+        self._set_peer_data("mme_addr", event.mme_ipv4_address)
+        logging.info(f"MME IPv4 address from LTE core: {event.mme_ipv4_address}")
+        self._configure_srsenb_service()
+        if self._get_peer_data("started"):
+            self.unit.status = MaintenanceStatus("Reloading srsenb.")
+            service_restart(SRS_ENB_SERVICE)
+            logging.info("Restarting EnodeB after MME IP address change.")
```

Comment:
How about logging the new MME IP? Something like:
```suggestion
            logging.info(f"Restarting EnodeB after MME IP address change. Current: {self._mme_addr}")
```

Reply:
Done